src/components/PunsForm/PunsForm.test.js

```diff
@@ -0,0 +1,136 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { PunsForm } from "./PunsForm";
+
+describe("<PunsForm />", () => {
+  describe("should render all elements", () => {
+    it("renders title", () => {
+      render(<PunsForm />);
+
+      const title = screen.getByRole("heading", { name: /add new pun/i });
+      expect(title).toBeInTheDocument();
+    });
+
+    it("renders date input", () => {
+      render(<PunsForm />);
+
+      const dateInput = screen.getByLabelText(/date:/i);
+
+      expect(dateInput).toBeInTheDocument();
+    });
+
+    it("renders dev input", () => {
+      render(<PunsForm />);
+
+      const devInput = screen.getByLabelText(/dev:/i);
+
+      expect(devInput).toBeInTheDocument();
+    });
+
+    it("renders context input", () => {
+      render(<PunsForm />);
+
+      const contextInput = screen.getByLabelText(/context:/i);
+
+      expect(contextInput).toBeInTheDocument();
+    });
+
+    it("renders pun input", () => {
+      render(<PunsForm />);
+
+      const punInput = screen.getByLabelText(/pun:/i);
+
+      expect(punInput).toBeInTheDocument();
+    });
+
+    it("renders button", () => {
+      render(<PunsForm />);
+
+      const addButton = screen.getByRole("button", { name: /add pun/i });
+
+      expect(addButton).toBeInTheDocument();
+    });
+  });
+  describe("should submit form when all fields are filled", () => {
+    it("calls the handleDate function", () => {
+      const handleDate = jest.fn();
+
+      render(<PunsForm handleDate={handleDate} />);
+
+      const dateInput = screen.getByLabelText(/date:/i);
+
+      userEvent.change(dateInput, { target: { value: "01/01/2020" } });
```

Comment:
```suggestion
      userEvent.type(dateInput, '2022-09-09');
```

Para evitarmos passar esse sufoco de novo, se lembrar de qd usarmos o userEvent.type c input type "date"  passar a data no formato correto YYYY-MM-DD

referência: https://github.com/testing-library/user-event/issues/941#issuecomment-1234374123